gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/branch/RepeatTest.java

```diff
@@ -386,5 +456,26 @@ public void g_V_hasXname_markoX_repeatXoutE_inV_simplePathX_untilXhasXname_rippl
         public Traversal<Vertex, Path> get_g_V_hasXloop_name_loopX_repeatXinX_timesX5X_path_by_name() {
             return g.V().has("loops","name","loop").repeat(__.in()).times(5).path().by("name");
         }
+
+        @Override
+        public Traversal<Vertex, Path> get_g_V_repeatXout_repeatXoutX_timesX1XX_timesX1X_limitX1X_path_by_name() {
+            // NB We need to prevent the RepeatUnrollStrategy from applying to properly exercise this test as this traversal can be simplified
```

Comment:
Nice to know that `RepeatUnrollStrategy` works in this way. @GCHQResearcher1337 could you add nested `repeat()` traversals (as necessary) to this parameterized unit test:

https://github.com/apache/tinkerpop/blob/d3b6a340dbb9c89355c40089a5c083dc0dbd4c5b/gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/optimization/RepeatUnrollStrategyTest.java#L74

to validate that `RepeatUnrollStrategy` continues to behave properly?

Reply:
hmm - seems like we shouldn't have extra barriers. @dkuppitz any thoughts on how this should be handled?

Reply:
For now, an extra check seems good to me. In the long run, we should get rid of these NoOpBarrierStep injections altogether as it should only be done by `LazyBarrierStrategy` (that goes back to the whole DFS vs BFS discussion).